include/oneapi/dpl/internal/dynamic_selection_impl/scheduler_defs.h

```diff
@@ -0,0 +1,46 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_SCHEDULER_DEFS_H
+#define _ONEDPL_SCHEDULER_DEFS_H
+
+#pragma once
+
+namespace oneapi {
+namespace dpl{
+namespace experimental{
+
+  template<typename NativeContext>
+  struct nop_execution_resource_t {
```

Comment:
I do not understand the purpose of this class. Could you please explain?

Reply:
The "nop" classes are for simplifying the implementation of wrapper types that have default, trivial behaviors.  We might be able to provide these defaults in other ways without these classes, but this is how it is now. These exist for execution resources (that do not need extra functions to act like execution resources -- e.g. you can already submit to them), property handles for policies that require no feedback, and selection handles that are straightforward wrappers around a single execution resource and single property handle.

Reply:
We will rename to clarify intent, along with all other nop classes.

Reply:
Replaced all nop_* classes with basic